dali/pipeline/data/buffer.h

```diff
@@ -322,6 +322,12 @@ class DLL_PUBLIC Buffer {
    */
   inline void set_type(const DALIDataType new_type_id) {
     DALI_ENFORCE(new_type_id != DALI_NO_TYPE, "new_type must be valid type.");
+    DALI_ENFORCE(type_.id() == new_type_id || type_.id() == DALI_NO_TYPE,
```

Comment:
Isn't it too strict? If it's about reallocations / changing the size in bytes, then the following would suffice:
```suggestion
    DALI_ENFORCE(type_.id() == new_type_id || size() == 0,
```

Reply:
Maybe a bit. Still, with the fact that we keep the Resize(shape, type) allowed, removing the set_type doesn't change that much, but maybe with the resize is just a case of naming or introducing another member function for that purpose, which we probably won't do soon.